tools/wasm-sourcemap.py

```diff
@@ -149,10 +150,29 @@ def read_dwarf_entries(wasm, options):
       file_path = (dir + '/' if dir != '' else '') + file.group(2)
       files[file.group(1)] = file_path
 
-    for line in re.finditer(r"\n0x([0-9a-f]+)\s+(\d+)\s+(\d+)\s+(\d+)", line_chunk):
-      entry = {'address': int(line.group(1), 16), 'line': int(line.group(2)), 'column': int(line.group(3)), 'file': files[line.group(4)]}
+    for line in re.finditer(r"\n0x([0-9a-f]+)\s+(\d+)\s+(\d+)\s+(\d+)(.*?end_sequence)?", line_chunk):
+      entry = {'address': int(line.group(1), 16), 'line': int(line.group(2)), 'column': int(line.group(3)), 'file': files[line.group(4)], 'eos': line.group(5) is not None}
       entries.append(entry)
 
+  # Remove dead functions' data. It is a heuristics to ignore data if the
+  # function starting address near to 0 (is equal to its size field length).
```

Comment:
Why is it 'near to 0' rather than just 0?   Sorry i know you tried to explain this to be me before but for some reason I'm not getting it.

Reply:
This offset is coming from Sym->Function->getFunctionCodeOffset() (see https://github.com/llvm-mirror/lld/blob/master/wasm/InputFiles.cpp#L151)

Reply:
Oh I see.  Should we change of offset to be the offset from the first instruction (not including the size field)?  I'm sorry if this is going back on territory we've already been over.

Do the downstream consumers care which form the offsets come in?  I suppose we chose the status quo so that it would make relocations easier to handle?

Reply:
The downstream consumers will only care where in code section or entire file the referred location is. Though making consumers to recognize that 0 address (and addresses derived from it) as dead function reference complicates their logic.

As I mentioned at https://reviews.llvm.org/D49446, we can just ignore function size field length -- it is a matter of changing 0xFDEAD000 to 0. Though problem of derived addressed stays and looks like consumers with have to these smarts (instead of just simple bounds check).

Reply:
P.S. By "smarts" I mean this exact patch, but without calculating function size field length and use =0 instead.